pkg/alertmanager/amcfg.go

```diff
@@ -135,7 +135,9 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, a
 	}
 
 	// Add routes to globalAlertmanagerConfig.Route without enforce namespace
-	globalAlertmanagerConfig.Route = cb.convertRoute(amConfig.Spec.Route, crKey)
```

Comment:
I think check convertedRoute nil is not necessary, it will not panic to assign a nil value :)